### PR TITLE
#37: updated get_list_of_unique_responses function

### PR DESCRIPTION
**Pull request by rebexxxxxx**
hi jake-- this is the update for the get_list_of_unique_responses function, allowing the function to take in a column of variables, not just 1 and using DISTINCT within the SQL query. i will address the other changes that are involved with this vector input in the next pull request.

### CODE REVIEW

**Review by jacobmpeters — COMMENTED** [review]
This looks good to me! Were you able to test it?

To test it, ensure that the output is identical regardless of whether the new function or the old function is used. Then make sure the notebook still runs.

Because this change will brake the function that calls it, I'll wait until you complete the other item before merging this. I think that your next push will automatically be added to this PR.

Let me know if you have questions.

**Comment by rebexxxxxx** [comment]
hi jake-- in this latest update, i further optimized the unique_responses() function to identify all unique responses from a vector of variables using 1 sql query. i also cleaned up the filter_vars_from_schema() function, removing a loop that called the unique_responses() function.

File: gcp_query_tools/connect_analytics_helper_functions.R [hunk 1]
```diff
@@ -245,52 +246,48 @@ add_novel_vars_to_csv <- function(vars, in_file, out_file = NULL) {
...
-    responses           <- df_var$var[[i]]
-    responses_available <- length(responses) > 0
-    if (responses_available){
-      for (j in 1:length(responses)){
-        response          <- responses[j]
-        response_is_novel <- !(response %in% unique_responses)
-        if (response_is_novel){
-          unique_responses <- append(unique_responses, response)
-        }
-      }
```
Comment (jacobmpeters) [inline gcp_query_tools/connect_analytics_helper_functions.R:285]: Glad to see this inefficient loop go!

File: gcp_query_tools/connect_analytics_helper_functions.R [hunk 2]
```diff
@@ -245,52 +246,48 @@ add_novel_vars_to_csv <- function(vars, in_file, out_file = NULL) {
...
+  # Initialize a string to store the SQL query
+  sql_query <- ""
+  
+  # Loop through each array variable to generate the SQL query
+  for (array_var in array_vars) {
+    
+    # Add to the SQL query a SELECT DISTINCT statement to get unique values for the array variable
+    sql_query <- glue("{sql_query}
+      SELECT DISTINCT
+        '{array_var}' AS variable,  # Label the array variable
```
Comment (jacobmpeters) [inline gcp_query_tools/connect_analytics_helper_functions.R:261]: This is an interesting way to construct the query in a loop. I'm not sure I understand it to be honest, but I know you've tested it. I might have you explain this to me sometime!

File: gcp_query_tools/connect_analytics_helper_functions.R [hunk 3]
```diff
@@ -245,52 +246,48 @@ add_novel_vars_to_csv <- function(vars, in_file, out_file = NULL) {
...
+  results_list <- split(as.numeric(df_var$array_element), df_var$variable)
+  
+  # Return the list of unique values for each variable
+  return(results_list)
 }
 
-filter_vars_from_schema <- function(project, table, schema, out_csv, out_json, 
+
+filter_vars_from_schema <- function(project, table, schema, out_csv, out_json, GUV_out_json,
                                     output_file_type = "json"
```
Comment (jacobmpeters) [inline gcp_query_tools/connect_analytics_helper_functions.R:289]: I have a few suggestions here:
- [ ] rename GUV_out_json to out_json and remove the redundant argument
- [ ] remove the output_file_type argument now that we no longer need optional file types of "json" or "js"

**Review by jacobmpeters — CHANGES_REQUESTED** [review]
Hi @rebexxxxxx , 

Thank you for writing such clean code and for testing it.

I'm just requesting a couple quick changes here related to the unnecessary arguments in the filter_vars_from_schema function. See related comment. 

Let me know if you have any questions!

**Pull request merged**